dds/idl/be_global.cpp

```diff
@@ -422,16 +424,49 @@ BE_GlobalData::parse_args(long& i, char** av)
     } else if (!ACE_OS::strncasecmp(av[i], NO_DCPS_DATA_TYPE_WARNINGS_FLAG, NO_DCPS_DATA_TYPE_WARNINGS_FLAG_SIZE)) {
       warn_about_dcps_data_type_ = false;
     } else if (!std::strcmp(av[i], "--default-extensibility")) {
-      if (!std::strcmp(av[i + 1], "final")) {
+      if (av[i + 1] == nullptr) {
```

Comment:
```suggestion
      if (av[++i] == 0) {
```

Same with other arguments.